xbps changes:

* Installs a "locale" service into the "boot" runlevel to set
  a global system locale.
* Adds consolefont by default in "boot" runlevel.
* Modified inittab to pass correct parameters to reboot(8).
* Sets unicode, rc_logger and rc_shell in /etc/rc.conf by default.
* Sets windowkeys and fix_euro in /etc/conf.d/keymaps by default.
* In local service, start/stop vmware-tools services if script exists.

--- runlevels/Makefile.Linux.orig	2009-05-07 15:48:37.075825332 +0000
+++ runlevels/Makefile.Linux	2009-05-07 15:48:52.008557039 +0000
@@ -1,3 +1,3 @@
 SYSINIT+=	devfs dmesg
-BOOT+=		hwclock keymaps modules mtab procfs termencoding
+BOOT+=		consolefont hwclock keymaps modules mtab procfs termencoding locale
 SHUTDOWN+=	killprocs mount-ro
--- init.d/Makefile.Linux.orig	2009-05-07 15:55:26.156910483 +0000
+++ init.d/Makefile.Linux	2009-05-07 15:55:35.682853819 +0000
@@ -2,7 +2,7 @@ NET_LO=	net.lo
 
 SRCS+=	devfs.in dmesg.in hwclock.in consolefont.in keymaps.in killprocs.in \
 	modules.in mount-ro.in mtab.in numlock.in procfs.in sysfs.in \
-	termencoding.in
+	termencoding.in locale.in
 
 .SUFFIXES:	.Linux.in

--- support/sysvinit/inittab.orig	2009-05-07 12:03:07.533948107 +0000
+++ support/sysvinit/inittab	2009-05-07 12:04:45.178070711 +0000
@@ -18,7 +18,7 @@ l3:3:wait:/sbin/rc default
 l4:4:wait:/sbin/rc default
 l5:5:wait:/sbin/rc default
 l6:6:wait:/sbin/rc reboot
-l6r:6:wait:/sbin/reboot -dk
+l6r:6:wait:/sbin/reboot -dfi
 #z6:6:respawn:/sbin/sulogin
 
 # new-style single-user
 .Linux.in:
--- conf.d/keymaps.orig	2009-10-09 17:09:02.537481085 +0200
+++ conf.d/keymaps	2009-10-09 17:12:19.694535631 +0200
@@ -1,10 +1,14 @@
 # Use KEYMAP to specify the default console keymap.  There is a complete tree
 # of keymaps in /usr/share/keymaps to choose from.
-keymap="us"
+#
+# Please note that is mandatory to also specify the keyboard layout model,
+# i.e qwerty, dvorak, etc; because otherwise loadkeys(1) will use the first
+# one it finds and some keymaps do have the same name.
+keymap="qwerty/us"
 
 # Should we first load the 'windowkeys' console keymap?  Most x86 users will
 # say "yes" here.  Note that non-x86 users should leave it as "no".
-windowkeys="NO"
+windowkeys="yes"
 
 # The maps to load for extended keyboards.  Most users will leave this as is.
 extended_keymaps=""
@@ -18,4 +22,4 @@ dumpkeys_charset=""
 
 # Some fonts map AltGr-E to the currency symbol ¤ instead of the Euro €
 # To fix this, set to "yes"
-fix_euro="NO"
+fix_euro="yes"
--- conf.d/staticroute.Linux.orig	2009-10-21 10:58:05.000000000 +0200
+++ conf.d/staticroute.Linux	2009-10-21 10:58:17.000000000 +0200
@@ -1,2 +1,3 @@
 # Example static route. See route(8) for syntax.
-staticroute="net 192.168.0.0 netmask 255.255.0.0 gw 10.73.1.1"
+#
+#staticroute="net 192.168.0.0 netmask 255.255.0.0 gw 10.73.1.1"
--- mk/prog.mk.orig	2009-10-21 11:00:28.000000000 +0200
+++ mk/prog.mk	2009-10-21 11:00:36.000000000 +0200
@@ -17,7 +17,6 @@ _DYNLINK_SH=		if test "${PREFIX}" = "" &
 			fi
 _DYNLINK!=		${_DYNLINK_SH}
 LDFLAGS+=		${_DYNLINK}$(shell ${_DYNLINK_SH})
-LDFLAGS+=		-Wl,-rpath=${PREFIX}/${LIBNAME}
 LDFLAGS+=		${PROGLDFLAGS}
 
 CLEANFILES+=		${OBJS} ${PROG}
--- mk/lib.mk.orig	2009-10-21 11:00:57.000000000 +0200
+++ mk/lib.mk	2009-10-21 11:01:17.000000000 +0200
@@ -32,7 +32,7 @@ lib${LIB}.a:	${OBJS} ${STATICOBJS}
 ${SHLIB_NAME}: ${VERSION_MAP}
 LDFLAGS+=	-Wl,--version-script=${VERSION_MAP}
 # We need to ensure we use libraries in /lib
-LDFLAGS+=	-L/${LIBNAME} -Wl,-rpath=/${LIBNAME}
+LDFLAGS+=	-L/${LIBNAME}
 
 ${SHLIB_NAME}:	${SOBJS}
 	@${ECHO} building shared library $@
--- conf.d/local.orig	2009-11-27 03:24:48.619351115 +0100
+++ conf.d/local	2009-11-27 03:26:21.519479673 +0100
@@ -5,6 +5,11 @@ local_start() {
 	# This is a good place to load any misc programs
 	# on startup (use &>/dev/null to hide output)
 
+	# Start vmware-tools services if script exists.
+	if [ -x /etc/init.d/vmware-tools ]; then
+		/etc/init.d/vmware-tools start
+	fi
+
 	# We should always return 0
 	return 0
 }
@@ -13,6 +18,11 @@ local_stop() {
 	# This is a good place to unload any misc.
 	# programs you started above.
 	
+	# Stop vmware-tools services if script exists
+	if [ -x /etc/init.d/vmware-tools ]; then
+		/etc/init.d/vmware-tools stop
+	fi
+	
 	# We should always return 0
 	return 0
 }
--- etc/rc.conf.in.orig	2009-12-05 03:30:48.675483763 +0100
+++ etc/rc.conf.in	2009-12-05 03:31:13.034222373 +0100
@@ -18,7 +18,7 @@
 # If not specified we use $SHELL, otherwise the one specified in /etc/passwd,
 # otherwise /bin/sh
 # Linux users could specify /sbin/sulogin
-#rc_shell=/bin/sh
+rc_shell=/sbin/sulogin
 
 # Do we allow any started service in the runlevel to satisfy the depedency
 # or do we want all of them regardless of state? For example, if net.eth0
@@ -44,7 +44,7 @@
 # /var/log/rc.log
 # NOTE: Linux systems require the devfs service to be started before
 # logging can take place and as such cannot log the sysinit runlevel.
-#rc_logger="YES"
+rc_logger="YES"
 
 # By default we filter the environment for our running scripts. To allow other
 # variables through, add them here. Use a * to allow all variables through.
@@ -73,7 +73,7 @@
 # There variables are shared between many init scripts
 
 # Set unicode to YES to turn on unicode support for keyboards and screens.
-#unicode="NO"
+unicode="YES"
 
 # Network fstypes. Below is the default.
 net_fs_list="afs cifs coda davfs fuse fuse.sshfs gfs glusterfs lustre ncpfs nfs nfs4 ocfs2 shfs smbfs"
